### ISSUE TITLE
Info window from redshift-gtk glitches out after closing with the 'x' on the window decoration

### ISSUE THREAD

**Issue by tostiheld** [jonls/redshift#220]
Hi,

When I open the info window from the redshift-gtk icon and then close it with the 'x' instead of the 'close' button it glitches out. The next time I try to open the info window there only appears a very very small window with no information. The glitched window is also not resizable.
When I close the window with the 'close' button the bug does not appear.

Steps to reproduce:
1. run redshift-gtk
2. right click on the appindicator icon and choose 'info'
3. close the info window with the 'x'
4. right click on the appindicator icon and choose 'info'

Expected result:
Another info window

Actual result:
A tiny non-resizable window with no information on it.

I run Debian Jessie with Openbox and tint2. My python version is 2.7.9.

**Comment by genodeftest** [comment]
I can confirm that on Fedora 22 with python 2.7.9. on GNOME 3.16.
When opening the window for the first time I get this message on stdout:
```
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
```
When opening it for the second time I get this block of critical warnings printed 6 times:
```
(redshift-gtk:18578): Gtk-CRITICAL **: _gtk_container_get_border_width_set: assertion 'GTK_IS_CONTAINER (container)' failed

(redshift-gtk:18578): Gtk-CRITICAL **: gtk_container_set_border_width: assertion 'GTK_IS_CONTAINER (container)' failed

(redshift-gtk:18578): Gtk-CRITICAL **: _gtk_container_set_border_width_set: assertion 'GTK_IS_CONTAINER (container)' failed

(redshift-gtk:18578): Gtk-CRITICAL **: _gtk_box_get_spacing_set: assertion 'GTK_IS_BOX (box)' failed

(redshift-gtk:18578): Gtk-CRITICAL **: gtk_box_set_spacing: assertion 'GTK_IS_BOX (box)' failed

(redshift-gtk:18578): Gtk-CRITICAL **: _gtk_box_set_spacing_set: assertion 'GTK_IS_BOX (box)' failed

(redshift-gtk:18578): Gtk-CRITICAL **: gtk_box_set_spacing: assertion 'GTK_IS_BOX (box)' failed

(redshift-gtk:18578): Gtk-CRITICAL **: _gtk_container_get_border_width_set: assertion 'GTK_IS_CONTAINER (container)' failed

(redshift-gtk:18578): Gtk-CRITICAL **: gtk_container_set_border_width: assertion 'GTK_IS_CONTAINER (container)' failed

(redshift-gtk:18578): Gtk-CRITICAL **: _gtk_container_set_border_width_set: assertion 'GTK_IS_CONTAINER (container)' failed
```
The first occurrence of these warnings is followed by the same message as the single line above.

**Comment by tostiheld** [comment]
I forked the project and I'm working on a fix. I'm not very experienced with python, but i'm tryin' :')
I think the problem is that the window is destroyed with the 'x' button, but hidden with the 'close' button. 

I tried to intercept the destroy event or to hide the close button, but those two methods didn't work.

I extracted the lines to create the info dialog to a separate function, and i execute the function every time you click on info in the context menu. The dialog reappears, but not all of the labels work.

**Comment by agnivade** [comment]
Hi there, I am also suffering the same bug and would like to help out. Can you point me in the direction that I should look at to make the fix ?

**Comment by jonls** [comment]
Could this perhaps be fixed by using a plain GtkWindow instead of using the GtkDialog? At least the first warning that @genodeftest mentioned seems to be caused by the fact that we're using a GtkDialog without a parent window.

**Comment by humitos** [comment]
I can confirm this issue in Xubuntu 16.04.

```
[22:29] humitos: ~> dpkg -l redshift-gtk redshift geoclue-2.0 python3.5 python2.7
Deseado=desconocido(U)/Instalar/eliminaR/Purgar/retener(H)
| Estado=No/Inst/ficheros-Conf/desempaqUetado/medio-conF/medio-inst(H)/espera-disparo(W)/pendienTe-disparo
|/ Err?=(ninguno)/requiere-Reinst (Estado,Err: mayúsc.=malo)
||/ Nombre                      Versión            Arquitectura       Descripción
+++-===========================-==================-==================-============================================================
ii  geoclue-2.0                 2.4.1-1            amd64              geoinformation service
ii  python2.7                   2.7.12-1~trusty1   amd64              Interactive high-level object-oriented language (version 2.7
ii  python3.5                   3.5.2-2~16.01      amd64              Interactive high-level object-oriented language (version 3.5
ii  redshift                    1.10-5ubuntu1      amd64              Adjusts the color temperature of your screen
ii  redshift-gtk                1.10-5ubuntu1      all                Adjusts the color temperature of your screen with GTK+ integ
[22:29] humitos: ~> 
```

**Comment by jonls** [comment]
Should be fixed with #491.
